### PR TITLE
Pull all required headers to be signed.

### PR DESCRIPTION
**Pull request by LaurenSpiegel**
Fixes https://github.com/scality/S3/issues/120

### CODE REVIEW

**Comment by alexandre-merle** [comment]
Related to S3 open issue https://github.com/scality/S3/issues/120 :)

File: lib/auth/auth.js [hunk 1]
```diff
@@ -117,11 +116,12 @@ auth.generateV4Headers =
...
         request.setHeader('x-amz-content-sha256', payloadChecksum);
-        Object.assign(request.headers, { host: request.getHeader('host') });
-        Object.assign(request.headers, { 'x-amz-date': amzDate });
-        Object.assign(request.headers,
-                      { 'x-amz-content-sha256': payloadChecksum });
-
+        Object.assign(request.headers, request._headers);
+        const signedHeaders = Object.keys(request._headers)
+            .filter(headerName =>
+            headerName.startsWith('x-amz-') || headerName.startsWith('x-scal')
```
Comment (alexandre-merle) [inline lib/auth/auth.js]: ```x-amz``` is enought since it's ```x-amz-scal``` for custom headers :)

File: lib/auth/auth.js [hunk 2]
```diff
@@ -117,11 +116,12 @@ auth.generateV4Headers =
         request.setHeader('host', request._headers.host);
         request.setHeader('x-amz-date', amzDate);
         request.setHeader('x-amz-content-sha256', payloadChecksum);
-        Object.assign(request.headers, { host: request.getHeader('host') });
-        Object.assign(request.headers, { 'x-amz-date': amzDate });
-        Object.assign(request.headers,
-                      { 'x-amz-content-sha256': payloadChecksum });
-
+        Object.assign(request.headers, request._headers);
```
Comment (alexandre-merle) [inline lib/auth/auth.js]: ```request.headers = request._headers;```

**Comment by alexandre-merle** [comment]
two minor comments :) :+1:

File: lib/auth/auth.js (hunk 1 above)
Comment (LaurenSpiegel) [inline lib/auth/auth.js]: I include both in the required headers test in case x-scal headers are created later.

Comment (alexandre-merle) [inline lib/auth/auth.js]: why not :)

File: lib/auth/auth.js (hunk 2 above)
Comment (LaurenSpiegel) [inline lib/auth/auth.js]: Causes a linting warning.

**Comment by LaurenSpiegel** [comment]
@DavidPineauScality and @AntoninCoulibaly, note this has a small impact on https://github.com/scality/Arsenal/pull/119

**Comment by LaurenSpiegel** [comment]
end to end passed http://ci.ironmann.io/gh/scality/Integration/2545

**Comment by MathieuCassagne** [comment]
LGTM :+1:

**Pull request merged**